stdlib/public/runtime/SwiftRT-ELF-WASM.cpp

```diff
@@ -48,9 +55,56 @@ static const void *__backtraceRef __attribute__((used, retain))
   BOUNDS_VISIBILITY extern const char __start_##name;   \
   BOUNDS_VISIBILITY extern const char __stop_##name;
 
+#if EMIT_NOTES
+/// A structure, compatible with the standard ELF note layout, that describes
+/// the bounds of a section known to Swift.
+///
+/// Sections described by these notes can be looked up at runtime using
+/// `dl_iterate_phdr()` unless an image's notes have been stripped.
+struct SectionNote {
+  /// The standard ELF note header.
+  ElfW(Nhdr) header;
+
+  /// The name of the ELF note.
+  ///
+  /// The size of this array must be a multiple of `sizeof(void *)` plus `4` to
+  /// ensure correct alignment on 64-bit archs (because `ElfW(Nhdr)` is 12 bytes
+  /// long and only 4-byte aligned.)
+  char n_name[36];
+
+  /// The "payload" of the note.
+  struct Bounds {
+    /// The start address of the section.
+    const void *start;
```

Comment:
Should (can?) these be relative pointers instead? I don't imagine it matters a whole lot either way, but we usually use relative pointers where we can.

Reply:
The base image address isn't statically knowable with ELF (`__dsohandle` is at an arbitrary offset) so we can't. But there are only going to be a handful of these, unlike with the metadata sections where we save many kilobytes.

Reply:
It would be self-relative, so you don’t need to obtain a different base address.

But yeah, the savings aren’t significant. I only bring this up out of a little foolish consistency.

Reply:
I suspect that would break the constness of it anyway. 🤷🏻‍♂️

Reply:
You're right, that expression would require a global initializer, so that would be no good.

Reply:
Lowering this to a compiler emission (which would be part of a "custom program header" proposal, I imagine) would allow for it, so let's keep that in mind for the future.

And there's nothing stopping us from emitting a custom program header _and_ notes, if we wanted to.

Reply:
Ugh I think I said that 3 different ways so far. Coffee before comments, folks.